specification/monitor/resource-manager/Microsoft.Insights/preview/2021-05-01-preview/diagnosticsSettings_API.json

```diff
@@ -0,0 +1,454 @@
+            "description": "Successful request to create a diagnostic setting",
+            "schema": {
+              "$ref": "#/definitions/DiagnosticSettingsResource"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Creates or Updates the diagnostic setting": {
+            "$ref": "./examples/createOrUpdateDiagnosticSetting.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "DiagnosticSettings"
+        ],
+        "operationId": "DiagnosticSettings_Delete",
+        "description": "Deletes existing diagnostic settings for the specified resource.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceUriParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/NameParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Successful request to remove a diagnostic setting"
+          },
+          "204": {
+            "description": "Successful request to remove a diagnostic setting"
+          }
+        },
+        "x-ms-examples": {
+          "Deletes the diagnostic setting": {
+            "$ref": "./examples/deleteDiagnosticSetting.json"
+          }
+        }
+      }
+    },
+    "/{resourceUri}/providers/microsoft.insights/diagnosticSettings": {
+      "get": {
+        "tags": [
+          "DiagnosticSettings"
+        ],
+        "operationId": "DiagnosticSettings_List",
+        "description": "Gets the active diagnostic settings list for the specified resource.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceUriParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Successful request to get more information about diagnostic setting",
+            "schema": {
+              "$ref": "#/definitions/DiagnosticSettingsResourceCollection"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        },
+        "x-ms-examples": {
+          "Gets the diagnostic setting": {
+            "$ref": "./examples/listDiagnosticSettings.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ProxyOnlyResource": {
```

Comment:
Consider referencing the one in common-types.

Reply:
Couldn't find in common types, will keep it.

Reply:
[https://github.com/Azure/azure-rest-api-specs/blob/297c0866ffaca818bd42ba20ea5d22f850aa4e18/specification/common-types/resource-management/v2/types.json#L82](https://github.com/Azure/azure-rest-api-specs/blob/297c0866ffaca818bd42ba20ea5d22f850aa4e18/specification/common-types/resource-management/v2/types.json#L82)